### PR TITLE
Add comment frontend

### PR DESCRIPTION
**Pull request by tin0d0**
- add comment component
- add comment sass

### CODE REVIEW

**Review by max-friedman — APPROVED** [review]
lgtm!

**Review by g-luo — COMMENTED** [review]
Same feedback as #58.

File: project/src/main/webapp/components/comment-component.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    this.time = '';
+  }
+
+  // Remove shadow DOM so styles are inherited
+  createRenderRoot() {
+    return this;
+  }
+
+  getServletData() {
+    fetch('/UserHome').then((response) => response.json()).then((documentsData) => {
```
Comment (tin0d0) [inline project/src/main/webapp/components/comment-component.js]: This is a place holder to show what comments are suppose to look like, servlet data will not be needed in the final comment-component

File: project/src/main/webapp/document.jsp [hunk 2]
```diff
@@ -177,6 +187,173 @@
...
+        }
+
+        // Create comment getElementsByName
+        document.getElementById('comment-container').innerHTML += '<comment-component name="<%= user.getNickname() %>" hash="<%= (String)request.getAttribute("documentHash") %>" date="' + date + '"></comment-component>';
+        document.querySelector('comment-component').firepad = firepad;
+        document.querySelector('comment-component').codeMirror = codeMirror;
+      }
+
+      // When a comment is submitted, pass the data to the Comment servlet
+      function subComment() {
```
Comment (g-luo) [inline project/src/main/webapp/document.jsp:215]: rename to submitComment, which is more descriptive

File: project/src/main/webapp/document.jsp [hunk 3]
```diff
@@ -177,6 +187,173 @@
...
+          // Highlight the text and give it the id of the associated comment
+          codeMirror.markText({line: startMarker.line, ch: startMarker.ch-1}, {line: endMarker.line, ch: endMarker.ch+2}, {className: "comment " + startMarker.id});
+
+          // Make the start and end markers read only so that the user doesn't accidentally delete them
+          codeMirror.markText({line: endMarker.line, ch: endMarker.ch}, {line: endMarker.line, ch: endMarker.ch+2}, {readOnly: true});
+          codeMirror.markText({line: startMarker.line, ch: startMarker.ch-1}, {line: startMarker.line, ch: startMarker.ch+1}, {readOnly: true});
+        }
+
+        // Load comments themselves
+        var hash = "<%= (String)request.getAttribute("documentHash") %>"
```
Comment (g-luo) [inline project/src/main/webapp/document.jsp:287]: put this in a new function to modularize

**Review by g-luo — COMMENTED** [review]
lgtm; good work with the commenting functions!

File: project/src/main/webapp/document.jsp (hunk 3 above)
Comment (tin0d0) [reply project/src/main/webapp/document.jsp:287]: sounds good

File: project/src/main/webapp/document.jsp (hunk 2 above)
Comment (tin0d0) [reply project/src/main/webapp/document.jsp:215]: will do

**Review by HidyHan — APPROVED** [review]

File: project/src/main/webapp/document.jsp (hunk 2 above)
Comment (frankpzn) [reply project/src/main/webapp/document.jsp:215]: +1

**Review by frankpzn — APPROVED** [review]

**Pull request merged**